GVFS.sln

```diff
@@ -263,9 +263,11 @@ Global
 		{03769A07-F216-456B-886B-E07CAF6C5E81}.Debug.Mac|x64.ActiveCfg = Debug|x64
 		{03769A07-F216-456B-886B-E07CAF6C5E81}.Debug.Mac|x64.Build.0 = Debug|x64
 		{03769A07-F216-456B-886B-E07CAF6C5E81}.Debug.Windows|x64.ActiveCfg = Debug|x64
+		{03769A07-F216-456B-886B-E07CAF6C5E81}.Debug.Windows|x64.Build.0 = Debug|x64

```

Comment:
`GVFS.Service.Mac.csproj` is now a dependency for `GVFS.UnitTests` project. This line makes sure `GVFS.Service.Mac.csproj` gets built on Windows as well.

Reply:
Why is that a dependency? Shouldn't it be that `GVFS.UnitTests.Mac` has the tests for `GVFS.Service.Mac`?

Reply:
`RepoRegistryTests.cs` used to be a `Windows` only test in `GVFS.UnitTests.Windows` project. Since `GVFS.Service.Mac` is now available, I moved `RepoRegistryTests.cs` to `GVFS.UnitTests` project, so that it gets run on both `Windows` and `Mac`.  There are 3 separate UT projects now. Trying to summarize the dependencies in the following table

| UT Project  | Test file included | SUT  | UT Project dependent on |
| ------------- | ------------- | ------------- |------------- |
| GVFS.UnitTests.Mac (**netcore**)  | `GVFS.UnitTests\Service\RepoRegistryTests.cs`  |`GVFS.Service\RepoRegistry.cs` |`GVFS.Service.Mac` (**netcore**) |
| GVFS.UnitTests.Windows (**framework**)  | `GVFS.UnitTests\Service\RepoRegistryTests.cs`  |`GVFS.Service\RepoRegistry.cs` |`GVFS.Service.Windows` (**framework**) |
| GVFS.UnitTests (**netcore**)| `GVFS.UnitTests\Service\RepoRegistryTests.cs`  |`GVFS.Service\RepoRegistry.cs` |`GVFS.Service.Mac` (**netcore**) |

This follows the existing pattern (@wilbaker correct me if I'm wrong) where `GVFS.UnitTests` which is a `.netcore` project uses other Mac projects like `GVFS.Platform.Mac`, `PrjFSLib.Mac.Managed` to satisfy its `.netcore` dependencies.

Alternate option is to move `RepoRegistry` (and its dependent classes like `RepoRegistration`) to `GVFS.Common` project.  `GVFS.Common`  is a dependency for all three UT projects plus also `GVFS.Service.Windows` and `GVFS.Service.Mac`.

Reply:
Wondering whether to keep the existing pattern, or move `RepoRegistry` classes to `GVFS.Common` in this PR or create a follow-up issue to clean this up. @derrickstolee @wilbaker @jrbriggs  please share your thoughts.

Reply:
CC: @nickgra

> This follows the existing pattern (@wilbaker correct me if I'm wrong) where GVFS.UnitTests which is a .netcore project uses other Mac projects like GVFS.Platform.Mac, PrjFSLib.Mac.Managed to satisfy its .netcore dependencies.

@alameenshah, is there something in the `MacServiceTests` tests that would not work if the tests were run on Windows?

`GVFS.UnitTests.Windows ` exists because there are some unit tests that require .NET Framework (which is only available on Windows).

All of the Mac unit tests have historically lived in `GVFS.UnitTests ` and been runnable on Windows because they have no platform dependencies and do not require .NET Framework.  By allowing the Mac unit tests to run on Windows it makes it easier to catch changes that would break Mac while developing on Windows.

I'm wondering why we need `GVFS.UnitTests.Mac`, what prevents us from just adding the new tests to `GVFS.UnitTests`?

Reply:
@alameenshah @nickgra I was finally able to successfully remove GVFS.UnitTest.Mac *and* keep the new unit tests.

Please see commit here (on top of this branch):

https://github.com/wilbaker/VFSForGit/commit/045250d031641cc7c1dc0307f83d8cbaaa7d9f5e

Getting this working made me realize the tests would probably be a lot cleaner if the common service code were moved into a new shared assembly (e.g. GVFS.Service.Common).

As it is now, it's a bit wonky because we're pulling in RepoRegistry from GVFS.Service.Mac (which makes it look Mac specific) when really it's common.  By moving it to a common assembly we can have confidence that our unit tests are covering the RepoRegistry used on both platforms.  If you agree, could you file a follow up work item for this?

Please let me know your feedback on my changes and if you have any questions on why I made the changes I did.

Thanks!

Reply:
Thank you @wilbaker. Have incorporated your changes in this PR. Agree with your comments about simplifying this further by consolidating common service code. Created a [tracking issue](https://github.com/Microsoft/VFSForGit/issues/1088).